lib/activerecord_autoreplica.rb

```diff
@@ -91,12 +91,17 @@ def self.using_read_replica_pool(replica_connection_pool)
   def self.in_replica_context(read_pool)
     return yield if current_read_pool # This method should not be reentrant
 
-    begin
-      self.current_read_pool = read_pool
-      yield
-    ensure
-      current_read_pool.release_connection
-      clear_current_read_pool
+    Thread.handle_interrupt(Exception => :never) do
+      setup_connection!
```

Comment:
Why does this need to be called here? The connection should be set up when the railtie is loaded

Reply:
It does, but:

a) Nothing guarantees the Railtie gets loaded, and
b) It doesn't get loaded in tests, which were failing

Reply:
It's surprising it doesn't get loaded in tests. But still not sure we should call it on every invocation rather then making sure it is called when the app is initialised.

Reply:
Well, calling it on all invocations is what the original code did — and adding it back fixed the test suite. It's a pretty cheap call, too.